source/agora/node/Ledger.d

```diff
@@ -125,37 +125,72 @@ public class Ledger
             throw new Exception("Genesis block loaded from disk is " ~
                 "different from the one in the config file");
 
-        // need to regenerate the UTXO set, starting from the genesis block
+        // need to rebuild the UTXO set and Validator set,
+        // starting from the genesis block
         if (this.utxo_set.length == 0)
         {
             Block block;
+            Block last_read_block = gen_block;
+
+            ManagedDatabase.beginBatch();
+            scope (failure) ManagedDatabase.rollback();
```

Comment:
Interesting. I think this was a good decision. 👍

Reply:
Good features should be used. Thanks @AndrejMitrovic